benchmark/sampling/neighborloader_bench.py

```diff
@@ -0,0 +1,105 @@
+import argparse
+import os.path as osp
+from timeit import default_timer
+
+from ogb.nodeproppred import PygNodePropPredDataset
+
+import torch_geometric.transforms as T
+from torch_geometric.datasets import OGB_MAG
+from torch_geometric.loader import NeighborLoader
+
+
+def run(args: argparse.ArgumentParser) -> None:
+
+    print("BENCHMARK STARTS")
+    for dataset_name in args.datasets:
+        print("Dataset: ", dataset_name)
+
+        if args.datasets_root == 'data':
+            root = osp.join(osp.dirname(osp.realpath(__file__)), '../..',
+                            'data',
+                            dataset_name.partition("-")[2])
+        else:
+            root = args.datasets_root
+
+        if dataset_name == 'ogbn-mag':
+            transform = T.ToUndirected(merge=True)
+            dataset = OGB_MAG(root=root, transform=transform)
+            train_idx = ('paper', dataset[0]['paper'].train_mask)
+            valid_idx = ('paper', dataset[0]['paper'].val_mask)
+            neighbour_sizes = args.hetero_neighbour_sizes
+        else:
+            dataset = PygNodePropPredDataset(dataset_name, root)
+            split_idx = dataset.get_idx_split()
+            train_idx = split_idx['train']
+            valid_idx = split_idx['valid']
+            neighbour_sizes = args.homo_neighbour_sizes
+
+        data = dataset[0].to(args.device)
+
+        print('Train sampling')
+        for sizes in neighbour_sizes:
+            print(f'Sizes={sizes}')
+            for batch_size in args.batch_sizes:
+                train_loader = NeighborLoader(data, num_neighbors=sizes,
+                                              input_nodes=train_idx,
+                                              batch_size=batch_size,
+                                              shuffle=True,
+                                              num_workers=args.num_workers)
+                start = default_timer()
+                iter = 0
+                times = []
+                for run in range(args.runs):
+                    start = default_timer()
+                    for batch in train_loader:
+                        iter = iter + 1
+                    stop = default_timer()
+                    times.append(round(stop - start, 3))
+                average_time = round(sum(times) / args.runs, 3)
+                print(f'Batch size={batch_size}   iterations={iter}   ' +
+                      f'times={times}    average_time={average_time}')
+        print('Validation sampling')
+        for batch_size in args.eval_batch_sizes:
+            val_loader = NeighborLoader(data, num_neighbors=[-1],
```

Comment:
Add some comment why this benchmark is needed/why it relates to validation sampling.

Reply:
Initially i thought about checking also some scenario with sampling all neighbors like more inference specific sampling? i changed the code - let me know if that make sense or there are some better real scenarios? WDYT?